src/metabase/models/params.clj

```diff
@@ -67,6 +70,38 @@
     :else
     field-id-or-form))
 
+(def ^:dynamic *ignore-current-user-perms-and-return-all-field-values*
+  "Whether to ignore permissions for the current User and return *all* FieldValues for the Fields being parameterized by
+  Cards and Dashboards. This determines how `:param_values` gets hydrated for Card and Dashboard. Normally, this is
+  `false`, but the public and embed versions of the API endpoints can bind this to `true` to bypass normal perms
+  checks (since there is no current User) and get *all* values."
+  false)
+
+(defn- field-ids->param-field-values-ignoring-current-user
+  "Given a list of field ids, return a map of field ids to all possible values for that field.
+  If a FieldValues for a field is not found, attempt to create it and return the created result."
+  [param-field-ids]
+  (let [fv-return-keys    [:values :human_readable_values :field_id]
+        field-id->fvs     (t2/select-fn->fn :field_id identity (cons FieldValues fv-return-keys)
+                                            :type :full
+                                            :field_id [:in param-field-ids])
+        missing-field-ids (set/difference (set param-field-ids) (set (keys field-id->fvs)))
+        missing-fields    (when (seq missing-field-ids)
+                            (t2/select Field :id [:in missing-field-ids]))]
+    (into field-id->fvs
+          (for [field missing-fields
+                :let [field-values (field-values/get-or-create-full-field-values! field)]]
```

Comment:
Does the `field-values/get-or-create-full-field-values!` function ignore the current user? It seems like it behaves that way but it just hits `qp/process-query` so I'm a bit surprised that it works if we aren't telling it to ignore the current user.